src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -710,35 +710,37 @@ private static CodeTypeBase ConvertComposedTypeToWrapper(CodeClass codeClass, Co
     protected static void AddDiscriminatorMappingsUsingsToParentClasses(CodeElement currentElement, string parseNodeInterfaceName, bool addFactoryMethodImport = false, bool addUsings = true) {
         if(currentElement is CodeMethod currentMethod &&
             currentMethod.Parent is CodeClass parentClass &&
-            parentClass.StartBlock is ClassDeclaration declaration) {
+            parentClass.StartBlock is ClassDeclaration declaration) { 
+            var parentClassNamespace = parentClass.GetImmediateParentOfType<CodeNamespace>();
                 if(currentMethod.IsOfKind(CodeMethodKind.Factory) &&
                     (parentClass.DiscriminatorInformation?.HasBasicDiscriminatorInformation ?? false)) {
-                        if(addUsings)
-                            declaration.AddUsings(parentClass.DiscriminatorInformation.DiscriminatorMappings
-                                .Select(static x => x.Value)
-                                .OfType<CodeType>()
-                                .Where(static x => x.TypeDefinition != null)
-                                .Select(x => new CodeUsing {
-                                    Name = x.TypeDefinition.GetImmediateParentOfType<CodeNamespace>().Name,
-                                    Declaration = new CodeType {
-                                        Name = x.TypeDefinition.Name,
-                                        TypeDefinition = x.TypeDefinition,
-                                    },
+                    if(addUsings)
+                        declaration.AddUsings(parentClass.DiscriminatorInformation.DiscriminatorMappings
+                            .Select(static x => x.Value)
+                            .OfType<CodeType>()
+                            .Where(static x => x.TypeDefinition != null)
+                            .Where(x => !x.TypeDefinition.GetImmediateParentOfType<CodeNamespace>().Name.Equals(parentClassNamespace.Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
Hi @baywet, looks like this line change is breaking a test in Go. It basically makes sure the class to be imported is not in the same namespace as the current class. Is this not a desirable thing to do in Go?

Reply:
go doesn't need to import symbols from the same package. So it's probably a case of unit test not setup properly.
Can you have a look and if you can't figure it out enlist Ronald to help you please?
Also reference comparison in this case is enough, no need to compare namespace names:
```suggestion
                            .Where(x => !x.TypeDefinition.GetImmediateParentOfType<CodeNamespace>() == parentClassNamespace.Name)
```

Reply:
Thanks for the feedback. I did replace it with.
```csharp
x.TypeDefinition.GetImmediateParentOfType<CodeNamespace>() != parentClassNamespace
```
instead of
```csharp
x.TypeDefinition.GetImmediateParentOfType<CodeNamespace>() != parentClassNamespace.Name
```
I guess that's what you meant.